hazelcast/src/main/java/com/hazelcast/query/impl/IndexService.java

```diff
@@ -56,11 +57,15 @@ public synchronized Index addOrGetIndex(String attribute, boolean ordered) {
     }
 
     public Index[] getIndexes() {
+        if (indexes.get() == null) {
```

Comment:
wouldn't be better to set the reference to EMPTY_INDEX in a constructor? you wouldn't have to deal with a special case.